crates/db_schema/src/lib.rs

```diff
@@ -157,6 +157,29 @@ pub enum PostListingMode {
   SmallCard,
 }
 
+#[derive(EnumString, Display, Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "full", derive(DbEnum, TS))]
+#[cfg_attr(
+  feature = "full",
+  ExistingTypePath = "crate::schema::sql_types::VoteDisplayModeEnum"
+)]
+#[cfg_attr(feature = "full", DbValueStyle = "verbatim")]
+#[cfg_attr(feature = "full", ts(export))]
+// TODO, in future releases, remove hide_scores, as this supercedes it.
+/// A vote-display setting that changes how votes are displayed in front ends.
+pub enum VoteDisplayMode {
```

Comment:
If anyone can think of any others, I'd be happy to add them.

Reply:
I would rather try to limit the number of options. Otherwise new apps will feel forced to implement all of them which can be a lot of work.

Reply:
Instead of an enum we could also consider representing it like this, which is more concise:

```rust
struct VoteDisplayMode {
    upvotes: bool,
    downvotes: bool,
    percentage: bool,
    score: bool
}
```

Reply:
Does having `vote_counts` enabled separately from `score` make sense UI wise? If there is a mix of up and down votes, the score will be the number of upvotes minus the number of downvotes. In other words, the score is derived from the vote counts.

Reply:
My main issue with having a lot of bools in that setting, is that 

1. Now we'd need to either add a lot of extra boolean columns, or extract them to another table and do a join just to get this info. Makes fetching an saving more complicated.
2. Some of them conflict, especially `HideAll`.

Reply:
The bools could be stored as different bits in a single u8 variable. I dont see how it would conflict, and `HideAll` is the same as having all bools false. And whats the problem with `DownvoteAndUpvotePercentage`, it would be the same as `vote_counts: true, percentage: true`.

Reply:
To cover all these so far, the struct would need to separate upvotes and downvotes, to be:

```rust
struct VoteDisplayMode {
    score: bool,
    upvotes: bool,
    downvotes: bool,
    upvote_percentage: bool,
}
```

As appealing as storing them all in a single column is, [it doesn't seem to be a good design pattern.](https://dba.stackexchange.com/questions/294111/is-there-a-common-pattern-for-storing-many-boolean-flags-other-than-adding-a-ton). Anyways I'll have to add other linked settings tables like `local_user_notifications`, so I think its fine to break this out into multiple tables, and name it `local_user_vote_display_mode` . 

We've had to do that for `local_site_rate_limit` anyway.

Reply:
I dont see anything in your link claiming that its a bad design pattern. The only problem I can imagine is that is difficult to read/modify individual booleans via sql, but we dont need that. And what do you mean by `local_user_notifications`?

Reply:
Its the top comment on [this answer](https://dba.stackexchange.com/a/294118/99741), and I agree with that.

When I start working on notifications, I'm also going to split this group of local user settings into its own table: `local_user_notifications`. So I might as well do the same for this one.